### PR TITLE
Fix join of MultiIndex to Index with one column and overlapping name.

### LINKED ISSUE #9823: [BUG] Joining MultiIndex with Int64Index does not find common columns
**Describe the bug**
Joining MultiIndex with Int64Index does not find common columns.

**Steps/Code to reproduce bug**
This doctest (#9815) fails:

https://github.com/rapidsai/cudf/blob/582cc6e466c7d941e1b34893fd56fbd42fe90d68/python/cudf/cudf/core/_base_index.py#L1100-L1115

The common column name `a` between the MultiIndex `lhs` and Int64Index `rhs` is not found. The following exception is raised:

```python
Failed example:
    lhs.join(rhs, how='inner')
Exception raised:
    Traceback (most recent call last):
      File "/home/bdice/code/compose/etc/conda/cuda_11.5/envs/rapids/lib/python3.8/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest BaseIndex.join[5]>", line 1, in <module>
        lhs.join(rhs, how='inner')
      File "/home/bdice/code/cudf/python/cudf/cudf/core/_base_index.py", line 1166, in join
        output = lhs._merge(rhs, how=how, on=on, sort=sort)
      File "/home/bdice/code/cudf/python/cudf/cudf/core/frame.py", line 3819, in _merge
        return merge_cls(
      File "/home/bdice/code/cudf/python/cudf/cudf/core/join/join.py", line 83, in __init__
        self._validate_merge_params(
      File "/home/bdice/code/cudf/python/cudf/cudf/core/join/join.py", line 413, in _validate_merge_params
        raise ValueError("No common columns to perform merge on")
    ValueError: No common columns to perform merge on
```

**Expected behavior**
Pandas matches the expected behavior of the docstring:

```python
>>> lhs = pd.DataFrame({"a": [2, 3, 1], "b": [3, 4, 2]}).set_index(["a", "b"]).index
>>> lhs
MultiIndex([(2, 3),
            (3, 4),
            (1, 2)],
           names=['a', 'b'])
>>> rhs = pd.DataFrame({"a": [1, 4, 3]}).set_index("a").index
>>> rhs
Int64Index([1, 4, 3], dtype='int64', name='a')
>>> lhs.join(rhs)
MultiIndex([(2, 3),
            (3, 4),
            (1, 2)],
           names=['a', 'b'])
>>> lhs.join(rhs, how="inner")
MultiIndex([(3, 4),
            (1, 2)],
           names=['a', 'b'])
```

### PR DESCRIPTION
**Pull request by vyasr**
This PR resolves #9823

### CODE REVIEW

**Comment by vyasr** [comment]
@shwina @bdice I believe I've fixed the underlying issue (the doctest @bdice linked on #9823 runs for me now and looks right) but I'd appreciate if one of you could add in a test then mark this as ready for review.

**Comment by vyasr** [comment]
For future reference: the issue is that the truthiness of a tuple `(None, )` is `True`, whereas `bool(None)` is `False`. The check needs to occur on the variable before it's wrapped in a non-empty container.

**Comment by vyasr** [comment]
The failure looks to be due to the test from #9808. Not sure why that would start failing unless something is out of sync, so I'll try rerunning once before investigating any further.

rerun tests

File: python/cudf/cudf/tests/test_joining.py [hunk 1]
```diff
@@ -2150,3 +2150,16 @@ def test_join_redundant_params():
...
+    # test joining a MultiIndex with an Index with overlapping name
+    lhs = (
+        cudf.DataFrame({"a": [2, 3, 1], "b": [3, 4, 2]})
+        .set_index(["a", "b"])
+        .index
+    )
+    rhs = cudf.DataFrame({"a": [1, 4, 3]}).set_index("a").index
+    expect = lhs.to_pandas().join(rhs.to_pandas(), how="inner")
+    got = lhs.join(rhs, how="inner")
+    assert_join_results_equal(expect, got, how="inner")
```
Comment (bdice) [inline python/cudf/cudf/tests/test_joining.py:2165]: This seems like an extraneous keyword argument.
```suggestion
    assert_join_results_equal(expect, got)
```

**Review by bdice — APPROVED** [review]
This fixes #9823 and the corresponding doctest in #9815. Thanks @vyasr @shwina!

**Comment by shwina** [comment]
@gpucibot merge

**Pull request merged**